src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,260 @@
-        case Success(Some(collectionType)) =>
-          val collectionMemberType = ModelSchema.getPlural(collectionType)
-          val entityData = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          val membershipData = TSVFormatter.makeMembershipTsvString(entities, entityType, collectionType, collectionMemberType.get)
-          val zipBytes: Array[Byte] = getZipBytes(entityType, membershipData, entityData)
-          val zippedFileName = entityType + ".zip"
-          RequestCompleteWithHeaders(
-            HttpEntity(ContentTypes.`application/octet-stream`, zipBytes),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> zippedFileName)))
-        case _ =>
-          val data = TSVFormatter.makeEntityTsvString(entities, entityType, attributeNames)
-          RequestCompleteWithHeaders(
-            (OK, data),
-            HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> filename)),
-            HttpHeaders.`Content-Type`(`text/plain`))
+
+  /*
+   * Helper Methods
+   */
+
+  // Standard exceptions have to be handled as a completed request
+  private def handleStandardException(ctx: RequestContext, t: Throwable): Future[Unit] = {
+    val errorReport = t match {
+      case f: FireCloudExceptionWithErrorReport => f.errorReport
+      case _ => ErrorReport(StatusCodes.InternalServerError, s"FireCloudException: Error generating entity download: ${t.getMessage}")
+    }
+    Future(ctx.complete(HttpResponse(
+      status = errorReport.statusCode.getOrElse(StatusCodes.InternalServerError),
+      entity = HttpEntity(ContentTypes.`application/json`, errorReport.toJson.compactPrint))))
+  }
+
+  // Stream exceptions have to be handled by directly closing out the RequestContext responder stream
+  private def handleStreamException(ctx: RequestContext, t: Throwable): Unit = {
+    val message = t match {
+      case f: FireCloudExceptionWithErrorReport => s"FireCloudException: Error generating entity download: ${f.errorReport.message}"
+      case _ => s"FireCloudException: Error generating entity download: ${t.getMessage}"
+    }
+    ctx.responder ! MessageChunk(message)
+    ctx.responder ! ChunkedMessageEnd
+  }
+
+  /**
+    * General Approach
+    * 1. Define a `Source` of entity queries
+    * 2. Run the source events through a `Flow`.
+    * 3. Flow sends events (batch of entities) to a streaming output actor
+    * 4. Return a Done to the calling route when complete.
+    */
+  private def streamSingularType(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, entityQueries: Seq[EntityQuery], metadata: EntityTypeMetadata, headers: IndexedSeq[String], attributeNames: Option[IndexedSeq[String]]): Future[Done] = {
+    // The output to the user
+    val streamingActorRef = actorRefFactory.actorOf(Props(new StreamingActor(ctx, ContentTypes.`text/plain`, entityType + ".txt")))
```

Comment:
This is pretty minor, but a usual Akka convention is to put a `props` method in the actor companion object, so you can instantiate it like this:
```
StreamingActor.props(ctx, ...)
```